library/src/G4CMPInterValleyScattering.cc

```diff
@@ -110,6 +119,56 @@ G4double G4CMPInterValleyScattering::GetMeanFreePath(const G4Track& track,
 G4VParticleChange* 
 G4CMPInterValleyScattering::PostStepDoIt(const G4Track& aTrack,
                                        const G4Step& aStep) {
+return (doValleySwitch ? SwitchValleys(aTrack, aStep) : ValleyScattering(aTrack, aStep));
+}
+
+
+// Deprecated IV scattering
+
+G4VParticleChange* 
+G4CMPInterValleyScattering::SwitchValleys(const G4Track& aTrack, 
+					 const G4Step& aStep) {
+  InitializeParticleChange(GetValleyIndex(aTrack), aTrack);
+  G4StepPoint* postStepPoint = aStep.GetPostStepPoint();
+  
+  if (verboseLevel > 1) {
+    G4cout << GetProcessName() << "::PostStepDoIt: Step limited by "
+	   << postStepPoint->GetProcessDefinedStep()->GetProcessName()
+	   << G4endl;
+  }
+  
+  // Don't do anything at a volume boundary
+  if (postStepPoint->GetStepStatus()==fGeomBoundary) {
+    return G4VDiscreteProcess::PostStepDoIt(aTrack, aStep);
+  }
```

Comment:
In principle, this code, which is duplicated in the other function, could be moved up to PostStepDoIt().  On the other hand, having it duplicated in each function will make the migration to G4CMP-476 (if we do that!) simpler.

I don't think you need to change anything.